### ISSUE TITLE
Boolean hyperparameter is 0/1 instead of False/True in hp.values dict

### ISSUE THREAD

**Issue by letmaik** [keras-team/keras-tuner#235]
I need to dump the values of all hyperparameters, and for that I'm accessing them as a dictionary using the `.values` field: https://github.com/keras-team/keras-tuner/blob/b7a5a1cd454429c4e0166ca3bea4241897e5e922/kerastuner/engine/hyperparameters.py#L464-L470

This works fine, however for boolean parameters, the value is 0 instead of False. Could this be made more consistent and return a bool?

**Comment by omalleyt12** [comment]
Please re-open if you can provide a repro

**Comment by rahulroynit** [comment]
Can you share an example that can help to understand the use of hp.Boolean for adding or leaving BatchNormalization()? Thanks!

**Comment by ag-tcm** [comment]
I also ran into this bug when using hp.Boolean with distributed tuning. Instead of the value being True/False it was 1/0 causing my program to through an error when I used that variable to define some part of my model. The workaround is to use hp.Int('param', 0, 1) and then wrap it around a function like:
def fixHPBool(val):
    return True if val == 1 else (False if val == 0 else val)
This way you'll always get a boolean value as expected until the bug is fixed.
